glnx-xattrs.c

```diff
@@ -87,40 +87,47 @@ read_xattr_name_array (const char *path,
   while (p < xattrs+len)
     {
       ssize_t bytes_read;
-      char *buf;
-      GBytes *bytes = NULL;
+      g_autofree char *buf = NULL;
+      g_autoptr(GBytes) bytes = NULL;
 
+    again:
       if (fd != -1)
         bytes_read = fgetxattr (fd, p, NULL, 0);
       else
         bytes_read = lgetxattr (path, p, NULL, 0);
       if (bytes_read < 0)
         {
+          if (errno == ENODATA)
+            continue;
```

Comment:
Since we're not advancing `p`, this is essentially the same as `goto again` right? Seems like things can still be racy and we can end up in an infinite loop if the xattr is removed between `[lf]listxattr` and `[lf]getxattr`.